scipy/spatial/transform/_rigid_transform.pyx

```diff
@@ -90,22 +97,26 @@ cdef _normalize_dual_quaternion(real_part, dual_part):
     prod_real, prod_dual = _compose_dual_quaternion(
         real_part, dual_part, real_conjugate, dual_conjugate)
 
+    # the squared norm of a dual quaternion is a dual number s + t * epsilon
+    s = prod_real[:, 3]
+    t = prod_dual[:, 3]
+
+    prod_real_norm = np.sqrt(np.maximum(s, 0.0))
```

Comment:
This maximum looks suspicious a little bit.

As far I understand `s` is equal to the sum of squares of the real part coefficients.

Not sure whether it's justified,  but for purity maybe it's better to remove it. If we get a negative number - something clearly went wrong and unexpectedly and needs to be investigated.

Reply:
I added it because I got an error in some of the test cases. Maybe this was caused by another error. We can try to remove it and see if the tests still work.

Reply:
OK, I removed the maximum. Seems to work fine in the tests. Maybe this was needed with the old version of the quaternion conjugate.